tools/ddrgen/src/scanners/dwarf/AixDwarfParser.cpp

```diff
@@ -151,31 +183,200 @@ isCompilerGenerated(const string data)
 {
 	return data.empty() ? false : ('c' == data[0]);
 }
+/**
+ * Parses data passed in as an array declaration
+ * param[in] data: the data to be parsed, should be in the format ar(INDEX_ID);(LOWER_BOUND);(UPPER_BOUND);(ARRAY_TYPE)
+ * param[out] currentDie: the DIE to populate with the parsed data.
+ */
+static int
+parseArray(const string data, Dwarf_Die currentDie, Dwarf_Error *error)
+{
+	int ret = DW_DLV_OK;
+	if (2 <= data.length()) {
+		if ("ar" == data.substr(0, 2)) {
+			str_vect arrayInfo = split (data, ';');
+			/*
+			 * arrayInfo will be in the form: ar(INDEX_ID);(LOWER_BOUND);(UPPER_BOUND);(ARRAY_TYPE)
+			 * arrayInfo[0] : ar + subrange type
+			 * arrayInfo[1] : lower bound
+			 * arrayInfo[2] : upper bound
+			 * arrayInfo[3] : array type
+			 */
+			/* Populate array */
+			Dwarf_Attribute type = new Dwarf_Attribute_s();
 
+			/* Create a DIE to hold info about the subrange type */
+			Dwarf_Die newDie = new Dwarf_Die_s();
+			Dwarf_Attribute subrangeType = new Dwarf_Attribute_s();
+			Dwarf_Attribute subrangeUpperBound = new Dwarf_Attribute_s();
+			if ((NULL == type) || (NULL == newDie) || (NULL == subrangeType) || (NULL == subrangeUpperBound)) {
+				ret = DW_DLV_ERROR;
+				setError(error, DW_DLE_MAF);
+			} else {
+				type->_type = DW_AT_type;
+				type->_form = DW_FORM_ref1;
+				type->_nextAttr = NULL;
+				/* If the type's typeID is built-in, leave it as is, otherwise, add the current file's ID to make it unique */
```

Comment:
This is an incorrect use of an attribute. When _form is DW_FORM_ref, only _ref can be set. _sdata should not be used unless the _form is DW_FORM_SDATA.

An attribute represents only one piece of information, such as a reference or file id, but never both.